arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -208,17 +208,23 @@ IResearchViewExecutor<ordered>::skipRows(size_t toSkip) {
   if (!_inputRow.isInitialized()) {
     if (_upstreamState == ExecutionState::DONE) {
       // There will be no more rows, stop fetching.
-      return {ExecutionState::DONE, stats, 0};
+      return std::make_tuple<ExecutionState, typename IResearchViewExecutor<ordered>::Stats, size_t>( // tupple, cannot use initializer list due to build failure
+        std::move(ExecutionState(ExecutionState::DONE)), std::move(stats), 0 // rvalue required for all elements to match constructor signature
+      );
     }
 
     std::tie(_upstreamState, _inputRow) = _fetcher.fetchRow();
 
     if (_upstreamState == ExecutionState::WAITING) {
-      return {_upstreamState, stats, 0};
+      return std::make_tuple<ExecutionState, typename IResearchViewExecutor<ordered>::Stats, size_t>( // tupple, cannot use initializer list due to build failure
+        std::move(ExecutionState(_upstreamState)), std::move(stats), 0 // rvalue required for all elements to match constructor signature
```

Comment:
```suggestion
        ExecutionState(_upstreamState), std::move(stats), 0 // rvalue required for all elements to match constructor signature
```

Reply:
I would think so too, but MSVC was complaining with:
cannot convert argument 1 from 'arangodb::aql::ExecutionState' to 'arangodb::aql::ExecutionState &&'

Reply:
Try without explicitly specifying the template argument types:
```cpp
return std::make_tuple(ExecutionState(_upstreamState), std::move(stats), 0);
```
`make_tuple` uses universal references and perfect forwarding, but this requires that the argument type is deduced by the compiler (i.e., not explicitly specified).

Reply:
@mpoeter removing the template arguments from std::make_tuple compiled everywhere, thank you